### PR TITLE
Removing player on click.

### PR DESCRIPTION
**Pull request by nibivi77**
### Description

This PR covers the use case where a player other than the host clicks "leave room" AND the booting functionality.

### CODE REVIEW

File: pages/room/[id].js [hunk 1]
```diff
@@ -146,6 +157,24 @@ export default function Room(props) {
   useEffect(() => {
     if (roomSubscription.data) {
       const { roomUpdated } = roomSubscription.data
+      // check to see if a user was deleted
+      users.forEach((user1) => {
+        const foundUser = roomUpdated.users.findIndex(
+          (user2) => user2.id === user1.id
```
Comment (DaceyTom2) [inline pages/room/[id].js:163]: Small consideration to rename user1 and user2 as it implies we are looking for 2 different users with the same id

File: pages/room/[id].js [hunk 2]
```diff
@@ -177,6 +206,51 @@ export default function Room(props) {
...
   }
+
+  const removeOnePlayer = async (playerId) => {
+    let playerIdToRemove = playerId || userId
+    const index = room.userIds.indexOf(playerIdToRemove)
+
+    if (index > -1) {
+      let copiedRoomUserIds = [...room.userIds]
+      copiedRoomUserIds.splice(index, 1)
+      console.log(copiedRoomUserIds)
```
Comment (DaceyTom2) [inline pages/room/[id].js:217]: Remove console log

File: pages/room/[id].js [hunk 3]
```diff
@@ -177,6 +206,51 @@ export default function Room(props) {
...
+        // remove user from backend
+        deleteUser({
+          variables: {
+            deleteUserId: playerIdToRemove,
+          },
+        }).catch((e) => {
+          throw e
+        })
+      } catch (e) {
+        console.log(e)
```
Comment (DaceyTom2) [inline pages/room/[id].js:238]: The app can handle this error

File: pages/room/[id].js [hunk 4]
```diff
@@ -177,6 +206,51 @@ export default function Room(props) {
...
+      }
+    }
+  }
+
+  const destroyRoom = () => {}
+
+  const leaveRoomClick = async () => {
+    if (userId === room.host) {
+      console.log(
```
Comment (DaceyTom2) [inline pages/room/[id].js:247]: I'm assuming this is a place holder for destroyRoom.
Either remove destroyRoom or call it here and console log the reminder from the placeholder function

File: pages/room/[id].js [hunk 5]
```diff
@@ -280,6 +354,7 @@ export default function Room(props) {
             isShown={room.isShown}
             currPlayer={getUserById(userId)}
             host={room.host}
+            onBootClick={removeOnePlayer}
```
Comment (DaceyTom2) [inline pages/room/[id].js:357]: Keep the naming consistent.
Either onClick or onBootClick/OnLeaveRoomClick

**Review by DaceyTom2 — CHANGES_REQUESTED** [review]

File: package.json [hunk 6]
```diff
@@ -30,6 +30,7 @@
     "prom-client": "^12.0.0",
     "prop-types": "^15.8.1",
     "react": "17.0.2",
+    "react-confirm-alert": "^2.8.0",
```
Comment (DaceyTom2) [inline package.json:33]: This dependency doesn't look promising based off use/issues/PRs https://github.com/GA-MO/react-confirm-alert
I'd flag this for a real app but is fine for prototypes

**Review by DaceyTom2 — APPROVED** [review]

**Review by MarcoGoC — APPROVED** [review]
Looks good, pending the package

**Pull request merged**